### ISSUE TITLE
AttributeError: 'FixtureClosureNode' object has no attribute 'remove' with pytest-cases

### ISSUE THREAD

**Issue by vbpatel73** [pytest-dev/pytest-bdd#380]
Hi,

my script runs ok but i am getting below error and because of it results scenario fail: can anyone help me pls?
```
venv\lib\site-packages\pytest_bdd\steps.py:232: in fin
    request._pyfuncitem._fixtureinfo.names_closure.remove(arg)
E   AttributeError: 'FixtureClosureNode' object has no attribute 'remove'
```

**Comment by The-Compiler** [comment]
Can you show the full output as well as your versions of Python, pytest and pytest-bdd?

**Comment by vbpatel73** [comment]
I have commented out below line 231 and 232 in steps.py and it works:

  # if add_fixturename:
        #     request._pyfuncitem._fixtureinfo.names_closure.remove(arg)

output shows tear down of scenario with 
venv\lib\site-packages\pytest_bdd\steps.py:232: in fin
request._pyfuncitem._fixtureinfo.names_closure.remove(arg)
E AttributeError: 'FixtureClosureNode' object has no attribute 'remove'

I set trace back short . i can make remove it and can produce full trace . pl let me know

**Comment by The-Compiler** [comment]
Yep, again, please show the full output and involved versions. Also please add [code tags](https://guides.github.com/features/mastering-markdown/) so that it doesn't get interpreted as markdown.

**Comment by vbpatel73** [comment]
please find full log

`============================================================== test session starts ==============================================================
platform win32 -- Python 3.7.6, pytest-5.3.2, py-1.8.0, pluggy-0.13.1
rootdir: C:\Automation\AGLC\Selenium\pytest_bdd, inifile: pytest.ini
plugins: allure-pytest-bdd-2.8.16, bdd-3.4.0, cases-1.16.0, excel-1.2.3, forked-1.1.3, html-2.0.1, json-report-1.2.1, metadata-1.8.0, parallel-0.0.10, reportlog-0.1.1, xdist-1.31.0
collected 20 items / 17 deselected / 3 selected                                                                                                  

pytest_bdd\tests\step_defs\test_Login.py .E.E.E                                                                                            [100%]

==================================================================== ERRORS =====================================================================
___________________________________________ ERROR at teardown of test_verify_invalid_login[browser0] ____________________________________________
aglc_venv\lib\site-packages\pytest_bdd\steps.py:232: in fin
    request._pyfuncitem._fixtureinfo.names_closure.remove(arg)
E   AttributeError: 'FixtureClosureNode' object has no attribute 'remove'
------------------------------------------------------------- Captured stdout call --------------------------------------------------------------
Login clicked
onx_aglc1@mailsac.com typed in UserName
1234 typed in Password
Login_btn clicked
Error: The Email Address or Password provided is incorrect. matches
____________________________________________ ERROR at teardown of test_verify_valid_login[browser0] _____________________________________________
aglc_venv\lib\site-packages\pytest_bdd\steps.py:232: in fin
    request._pyfuncitem._fixtureinfo.names_closure.remove(arg)
E   AttributeError: 'FixtureClosureNode' object has no attribute 'remove'
------------------------------------------------------------- Captured stdout call --------------------------------------------------------------
Login clicked
onx_aglc1@mailsac.com typed in UserName
Myname@20 typed in Password
Login_btn clicked
MyAccountclicked after wait
Logout clicked
__________________________________________ ERROR at teardown of test_verify_forgot_password[browser0] ___________________________________________
aglc_venv\lib\site-packages\pytest_bdd\steps.py:232: in fin
    request._pyfuncitem._fixtureinfo.names_closure.remove(arg)
E   AttributeError: 'FixtureClosureNode' object has no attribute 'remove'
------------------------------------------------------------- Captured stdout call --------------------------------------------------------------
Login clicked
forgot password clicked
onx_aglc1029@mailsac.com typed in EmailAddressForgotPwd
forgotPassword_btn clicked
Back To Login btn clicked by JS
waiting for page to load
--------------------------------------- generated html file: file://C:\Automation\AGLC\Selenium\bdd.html ----------------------------------------
============================================= 3 passed, 17 deselected, 3 errors in 74.71s (0:01:14) =============================================`

**Comment by The-Compiler** [comment]
For multiline-code, you will need to use three backticks (\`) on a single line before/after the code (or select the code and use the "<>" button). I edited your post accordingly and also changed the title to something easier to search for :slightly_smiling_face: 

This looks like an incompatibility with pytest-cases, as `FixtureClosureNode` is coming from there: https://github.com/smarie/python-pytest-cases/blob/2.0.4/pytest_cases/plugin.py#L116

Not sure what's going on exactly, though - cc @smarie

**Comment by vbpatel73** [comment]
yes you are right. I unistall pytest-cases and it works now.
thanks,

**Comment by smarie** [comment]
Hi @The-Compiler thanks for connecting :) 

It seems that pytest-bdd tries to remove a fixture from the closure. Since pytest-cases replaces the closure with a "super-closure" able to handle [alternatives induced by fixture unions](https://smarie.github.io/python-pytest-cases/unions_theory/), it is a bit less trivial to support appending and removing from the closure.

I opened a ticket: https://github.com/smarie/python-pytest-cases/issues/116

I will have a look

**Comment by The-Compiler** [comment]
Perfect, let's continue this there then!

**Comment by smarie** [comment]
This should be fixed in pytest-cases 2.1.0. @vbpatel73 can you please confirm ?
